### PR TITLE
Added API to allow creation of users

### PR DESCRIPTION
**Pull request by wgerlach**

### CODE REVIEW

File: services/beehive-upload-server/user-service/main.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	file, err := os.Open("/etc/passwd")
+
+	if err != nil {
+		err = fmt.Errorf("error opening file: %s", err.Error())
+		return
+	}
+
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
```
Comment (seanshahkarami) [inline services/beehive-upload-server/user-service/main.go:42]: You may like using bufio.NewScanner for this. It makes it easy to write a "for each line via":
```go
scanner := bufio.NewScanner(file)

for scanner.Scan() {
    // process line in scanner.Text()
}

if scanner.Err() != nil {
  handle / return scanner.Err()
}
```

File: services/beehive-upload-server/user-service/main.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	fmt.Fprint(w, string(response_json))
+
+}
+
+func CreateUser(username string) (err error) {
+	// adduser -D -g "" "$username"
+	// passwd -u "$username"
+	// chown -R "$username:$username" "/home/$username"
+	var out []byte
+	out, err = exec.Command("adduser", "-D", "-g", "\"\"", username).CombinedOutput()
```
Comment (seanshahkarami) [inline services/beehive-upload-server/user-service/main.go:114]: You may like the assign + check syntax for this, if you were worried about shadowing `out`. 

Something like:

```go
if out, err := exec.Command(args).CombinedOutput(); err != nil {
  // handle / return err
}
```

File: services/beehive-upload-server/user-service/main.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+	user = strings.TrimPrefix(user, "node-")
+
+	if len(user) != 16 {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprint(w, "invalid user name, wrong length")
+		return
+	}
+
+	re := regexp.MustCompile(`^[0-9a-f]{16}$`)
```
Comment (seanshahkarami) [inline services/beehive-upload-server/user-service/main.go:152]: You may want to make this a global
```go
var nodeIDPattern = regexp.MustCompile(...)
```

That way the program will panic right away if it's a bad regexp.

**Review by seanshahkarami — APPROVED** [review]
Looks good to me. I added a few very minor tweaks you may like

File: services/beehive-upload-server/user-service/main.go (hunk 1 above)
Comment (wgerlach) [reply services/beehive-upload-server/user-service/main.go:42]: thanks for the tip !

**Pull request merged**